Fixing UI on Settings/Preferences/Other for Language Translation Preferences

lacimonsrud opened the pull request:
Hi! user adamniedzielski orginially tried to implement issue #29989 with PR #31209 and the UI seemed a little wonky with having two large lists. I tried to fix the UI a bit by changing the orientation of the lists, while still maintaining the code that they originally implemented for the functionality for the users to select which languages they understand. I'm open to ideas on how to refine this UI. I wanted to make it so that the right hand list of checkboxes didn't contain any labels, but I couldn't figure out how. Thanks!
![image](https://github.com/user-attachments/assets/ceb98f01-228d-420e-b456-7f30adf967e4)

adamniedzielski committed:
Add a way for the user to select which languages they understand

Closes #29989

adamniedzielski committed:
Fix locale

adamniedzielski committed:
Fix jslint

adamniedzielski committed:
Fix typecheck

lacimonsrud committed:
Test

lacimonsrud committed:
Test2

lacimonsrud committed:
Made UI Changes to Settings/Preferences/Other

lacimonsrud committed:
Made UI Changes to Settings/Preferences/Other

mjankowski commented:
Can you add before/after screenshots?

Separately, I have same question here as on other one, which is I wonder about a boolean "Dont translate my selected languages" option, instead of repeating the list (understood they are not the same thing).

lacimonsrud commented:
Hi! So sorry, I initially intended to include screenshots, but forgot to do it before I submitted the PR. 
I read your previous comment about the boolean "Don't translate my selected languages" button, but it didn’t really make sense to me. I like the idea of the boolean button, but I’m curious how the users would then select languages? If the button is true, would the list then pop up to allow users to select languages? If the button is false, would the user have no option to select languages? Or would the button simply operate off of the filtered languages tab? I feel like that would cause some weird cases if we tried to overlap translating languages with filtering languages.

mjankowski commented:
> Hi! So sorry, I initially intended to include screenshots, but forgot to do it before I submitted the PR.

No problem -- might make it easier for review if you are able to add.

> I read your previous comment about the boolean "Don't translate my selected languages" button, but it didn’t really make sense to me.

It was a sort of half-baked brainstorm -- trying to avoid essentially repeating the huge language list in the UI.

> Or would the button simply operate off of the filtered languages tab? I feel like that would cause some weird cases if we tried to overlap translating languages with filtering languages.

Yes, that was the idea, and I agree there would be edge cases and awkwardness. It would work well for the narrow use case of "I can understand and do not want translated any of the languages I've selected for filtering", and it would break down a bit outside those use cases.

Other ideas:

- Replace the current thing with an autocompleting multi-select dropdown, add another one for translation
- Do one language list, have "filtering" and "translating" checkboxes separately next to them
- One list, with dropdown next to each one with "policies" or sorts? ("Exclude", "Include and translate", "Include and preserve original", etc sort of thing...)

Anyway -- screenshots might help, but also -- UX review capacity on team right now is almost zero, so no idea if will be seen.

lacimonsrud commented:
Just updated my original post with a screenshot of the version I implemented. Like I said in my original post, I thought it would work best if I could somehow remove the labels on the second column, so that it would just sort of list out all the languages once and then have checkboxes for the two different categories. I definitely like the idea of the multi-select list though, and I actually did attempt that originally, but it was kind of finicky with the selection.
